evm_arithmetization/src/cpu/kernel/asm/transactions/type_3.asm

```diff
@@ -154,7 +154,33 @@ store_origin:
     %eq_const(0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff)
     %jumpi(panic)
 
+    // stack: address, retdest
+
+    // EIP-4844: Deduct blob_gas_fee from the sender and burn it
+    %compute_blob_gas_fee
+    DUP2
+    // stack: address, blob_gas_fee, address, retdest
+    %deduct_eth
+    // stack: deduct_eth_status, address, retdest
+    %jumpi(transfer_eth_failure)
+
     // stack: address, retdest
     %mstore_txn_field(@TXN_FIELD_ORIGIN)
     // stack: retdest
     %jump(process_normalized_txn)
+
+%macro compute_blob_gas_fee
+    PUSH @GAS_PER_BLOB
+    %get_blob_versioned_hashes_list_length
+    MUL
+    PROVER_INPUT(blobbasefee)
```

Comment:
note: I'm reusing this here as we call PROVER_INPUT for the `BLOBBASEFEE` opcode, although we could implement it relatively cheaply in the KERNEL, if we were to want the switch.